pkg/channel/data.go

```diff
@@ -31,5 +30,5 @@ type DataChan struct {
 	// if not set then the data is sent to out channel and processed by side car  default method
 	OnReceiveOverrideFn func(e cloudevents.Event) error
 	// ProcessEventFn  Optional, this allows to customize message handler thar was received at the out channel
-	ProcessEventFn func(e event.Event) error
+	ProcessEventFn func(e interface{}) error
```

Comment:
will this slow down, by checking for type on every event we get ?
@atyronesmith what are your thoughts? trying to accommodate both hardware schema and cloud-native schema